wlauto/common/linux/workload.py

```diff
@@ -0,0 +1,148 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+from math import ceil
+import time
+
+from wlauto.core.extension import Parameter
+from wlauto.core.workload import Workload
+from wlauto.core.resource import NO_ONE
+from wlauto.common.resources import Executable
+from wlauto.common.android.resources import ReventFile
+from wlauto.exceptions import WorkloadError
+from wlauto.utils.revent import ReventRecording
+
+class ReventWorkload(Workload):
+    # pylint: disable=attribute-defined-outside-init
+
+    parameters = [
+        Parameter('default_naming', kind=bool, default=False,
+                  description='Use the default naming. Set this option to true if the revent file you wish to use was '
+                              'generated using the wa record command.'),
+        Parameter('idle', kind=bool, default=False,
+                  description='Say whether or not you want this workload to be run in idle mode. In idle mode the revent '
+                              'will only run .setup and .close files and in between will be idle for however many seconds '
+                              'you specify.'),
+        Parameter('idle_time', kind=int, default=10,
+                  description='The time you wish the device to remain idle for (idle must be True for this to be relevant).'),
+        Parameter('file_name', kind=str, default=None,
+                  description='The name of the revent file to look for. Will also look for file_name.setup and file_name.close'),
+        Parameter('no_setup', kind=bool, default=False,
+                  description='Set to True to force disable the setup revent file from running.'),
+        Parameter('no_close', kind=bool, default=False,
+                  description='Set to True to force disable the close revent file from running.'),
+    ]
+
+    def __init__(self, device, _call_super=True, **kwargs):
+        if _call_super:
+            Workload.__init__(self, device, **kwargs)
+        self.setup_timeout = kwargs.get('setup_timeout', None)
+        self.run_timeout = kwargs.get('run_timeout', None)
+        self.close_timeout = kwargs.get('close_timeout', None)
+        self.revent_setup_file = None
+        self.revent_run_file = None
+        self.revent_close_file = None
+        self.on_device_setup_revent = None
+        self.on_device_run_revent = None
+        self.on_device_close_revent = None
+        self.statedefs_dir = None
+
+    def initialize(self, context):
+        devpath = self.device.path
+        self.on_device_revent_binary = devpath.join(self.device.binaries_directory, 'revent')
+
+    def setup(self, context):
+        devpath = self.device.path
+        if not self.no_setup:
+            self.logger.info('Trying to get .setup')
```

Comment:
Have removed this from the ReventWorkload class and instead added a debug line for ReventGetter